apps/beeai-server/src/beeai_server/configuration.py

```diff
@@ -266,6 +284,33 @@ def _oci_registry_defaultdict(self):
             self.provider_build.oci_build_registry_prefix = next(iter(self.oci_registry.keys()))
         return self
 
+    @model_validator(mode="after")
+    def _github_registry_config(self):
+        try:
+            for registry, conf in self.github_registry_config_json.auths.items():
+                self.github_registry[registry] = conf
+        except ValueError as e:
+            logger.error(f"Failed to parse .githubconfigjson: {e}. GitHub access might not work correctly.")
+        return self
+
+    @model_validator(mode="after")
+    def _set_default_provider_build_values(self):
+        self.provider_build.k8s_namespace = self.provider_build.k8s_namespace or self.k8s_namespace
+        self.provider_build.k8s_kubeconfig = self.provider_build.k8s_kubeconfig or self.k8s_kubeconfig
+        self.provider_build.manifest_template_dir = (
+            self.provider_build.manifest_template_dir or self.provider.manifest_template_dir
+        )
+        return self
+
+    @model_validator(mode="after")
+    def _parse_kubeconfig(self):
```

Comment:
Having the same prop mean "path to kubeconfig file" and "dict of kubeconfig data" at different points is weird, wouldn't a cached property work better here?

Reply:
I made it the same hybrid type that kr8s accepts, so that I can load the secret as env rather than a file, because I thought there were more security implications with files, but I'll refactor it to accept just Path, I think there are no risks with mounting secrets as files after all.